tests/unit/ctl/test_branch_report.py

```diff
@@ -0,0 +1,436 @@
+    client = InfrahubClient(config=Config(address="http://mock"))
+    result = await check_git_files_changed(client, branch="test-branch")
+    assert result is False
+
+
+async def test_check_git_files_changed_empty_response(
+    mock_git_files_empty_response: HTTPXMock,
+) -> None:
+    """Test check_git_files_changed returns False when branch not in response."""
+    client = InfrahubClient(config=Config(address="http://mock"))
+    result = await check_git_files_changed(client, branch="test-branch")
+    assert result is False
+
+
+async def test_check_git_files_changed_http_error(httpx_mock: HTTPXMock) -> None:
+    """Test check_git_files_changed raises exception on HTTP error."""
+    httpx_mock.add_response(
+        method="GET",
+        url="http://mock/api/diff/files?branch=test-branch",
+        status_code=404,
+    )
+
+    client = InfrahubClient(config=Config(address="http://mock"))
+    with pytest.raises(HTTPStatusError):
+        await check_git_files_changed(client, branch="test-branch")
+
+
+@pytest.fixture
+def mock_branch_report_command(httpx_mock: HTTPXMock) -> HTTPXMock:
+    """Mock all responses for branch report CLI command."""
+    httpx_mock.add_response(
+        method="POST",
+        json={
+            "data": {
+                "Branch": [
+                    {
+                        "id": "test-branch-id",
+                        "name": "test-branch",
+                        "sync_with_git": True,
+                        "is_default": False,
+                        "origin_branch": "main",
+                        "branched_from": "2025-11-01T10:00:00Z",
+                        "has_schema_changes": False,
+                        "status": "OPEN",
+                    }
+                ]
+            }
+        },
+        match_headers={"X-Infrahub-Tracker": "query-branch"},
+    )
+
+    httpx_mock.add_response(
+        method="POST",
+        json={
+            "data": {
+                "DiffTree": {
+                    "num_added": 5,
+                    "num_updated": 3,
+                    "num_removed": 1,
+                    "num_conflicts": 0,
+                    "to_time": "2025-11-14T18:00:00Z",
+                    "from_time": "2025-11-01T10:00:00Z",
+                    "base_branch": "main",
+                    "diff_branch": "test-branch",
+                    "name": None,
+                    "nodes": [],
+                }
+            }
+        },
+    )
+
+    httpx_mock.add_response(
+        method="GET",
+        json={"test-branch": {"repo-1": {"files": [{"path": "test.py"}]}}},
+    )
+
+    httpx_mock.add_response(
+        method="POST",
+        json={"data": {"CoreProposedChange": {"count": 0, "edges": []}}},
+    )
+
+    return httpx_mock
+
+
+def test_branch_report_command_without_proposed_change(
+    mock_branch_report_command: HTTPXMock,
+    mock_schema_query_05,  # type: ignore[misc]
```

Comment:
Looks like this is a `HTTPXMock` class to, so it should be fine to add that and remove the ignore statement?